pkg/controller/plan/controller.go

```diff
@@ -210,9 +210,12 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 	// Validations.
 	err = r.validate(plan)
 	if err != nil {
+		if r.isDanglingArchivedPlan(plan) {
```

Comment:
It would probably be worthwhile to check what the specific error is before archiving, and the condition probably does not need to be hidden behind a method call.